src/Sylius/Component/Core/OrderProcessing/StateResolver.php

```diff
@@ -61,12 +61,6 @@ public function resolvePaymentState(OrderInterface $order)
      */
     public function resolveShippingState(OrderInterface $order)
     {
-        if ($order->isBackorder()) {
-            $order->setShippingState(OrderShippingStates::BACKORDER);
-
-            return;
-        }
-
         $order->setShippingState($this->getShippingState($order));
```

Comment:
State machine should be used for that. Take a look at https://github.com/Sylius/Sylius/pull/5605 and [my comment](https://github.com/Sylius/Sylius/pull/5605/files#r72582095).